### PR TITLE
JS: Some more improvements to d.ts file analysis

### PR DESCRIPTION
**Pull request by asgerf**
Some more improvements needed for model generation, mainly by improving analysis of `.d.ts` files.

[Evaluation](https://github.com/github/codeql-dca-main/tree/data/asgerf/madman-prep2__default__code-scanning__3/reports) looks OK

### CODE REVIEW

File: javascript/extractor/src/com/semmle/ts/extractor/TypeScriptASTConverter.java [hunk 1]
```diff
@@ -1710,7 +1710,10 @@ private Node convertNamespaceDeclaration(JsonObject node, SourceLocation loc) th
     }
     if (nameNode instanceof Literal) {
       // Declaration of form: declare module "X" {...}
-      return new ExternalModuleDeclaration(loc, (Literal) nameNode, body);
+      ExternalModuleDeclaration decl = new ExternalModuleDeclaration(loc, (Literal) nameNode, body);
+      attachSymbolInformation(decl, node);
+      System.out.println("ExternalModuleDeclaration symbol = " + decl.getSymbol());
```
Comment (erik-krogh) [inline javascript/extractor/src/com/semmle/ts/extractor/TypeScriptASTConverter.java:1715]: Debug print? Remove?

File: javascript/ql/lib/semmle/javascript/NPM.qll [hunk 2]
```diff
@@ -180,6 +180,35 @@ class PackageJson extends JsonObject {
   Module getMainModule() {
     result = min(Module m, int prio | m.getFile() = resolveMainModule(this, prio) | m order by prio)
   }
+
+  /**
+   * Gets the `types` or `typings` field of this package.
+   */
+  string getTypings() { result = this.getPropStringValue(["types", "typings"]) }
```
Comment (erik-krogh) [inline javascript/ql/lib/semmle/javascript/NPM.qll:187]: There is also the `"types`"  field under `"exports"`: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-rc/#package-json-exports-imports-and-self-referencing  

(Might not be that relevant, so you can probably skip it).

**Review by erik-krogh — COMMENTED** [review]
very quick first look. 
Seems reasonable, but two comments.

File: javascript/ql/lib/semmle/javascript/NPM.qll (hunk 2 above)
Comment (asgerf) [reply javascript/ql/lib/semmle/javascript/NPM.qll:187]: The new `exports`/`imports` fields are too complex to handle in this PR

**Review by erik-krogh — APPROVED** [review]

**Pull request merged**